src/opera/commands/notify.py

```diff
@@ -0,0 +1,141 @@
+
+from opera.commands.info import info
+from opera.error import DataError, ParseError
+from opera.parser import tosca
+from opera.storage import Storage
+from opera.utils import prompt_yes_no_question
+
+
+def add_parser(subparsers):
+    parser = subparsers.add_parser(
+        "notify",
+        help="Notify the orchestrator about the changes (after the deployment) and run triggers from TOSCA policies"
+    )
+    parser.add_argument(
+        "--instance-path", "-p",
+        help="Storage folder location (instead of default .opera)"
+    )
+    parser.add_argument(
+        "--inputs", "-i", type=argparse.FileType("r"),
+        help="YAML or JSON file with inputs to override the inputs supplied in deploy",
+    ).complete = shtab.FILE
+    parser.add_argument(
+        "--trigger", "-t",
+        help="TOSCA policy trigger name or event that will invoke all the actions (interface operations) on policy",
+    )
+    parser.add_argument(
+        "--notification", "-n", type=argparse.FileType("r"),
+        help="Notification file (usually JSON) with changes that will be exposed to TOSCA interfaces",
+    ).complete = shtab.FILE
+    parser.add_argument(
+        "--force", "-f", action="store_true",
+        help="Force the action and skip any possible prompts",
+    )
+    parser.add_argument(
+        "--verbose", "-v", action="store_true",
+        help="Turns on verbose mode",
+    )
+    parser.set_defaults(func=_parser_callback)
+
+
+def _parser_callback(args):
+    if args.instance_path and not path.isdir(args.instance_path):
+        raise argparse.ArgumentTypeError("Directory {} is not a valid path!".format(args.instance_path))
+
+    storage = Storage.create(args.instance_path)
+    status = info(None, storage)["status"]
+
+    if not args.force and storage.exists("instances"):
+        if status == "initialized":
+            print("The nodes have not been deployed yet and running notify might have unexpected consequences.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+        if status == "interrupted":
+            print("The deployment has been interrupted and running notify might have "
+                  "unexpected consequences on the nodes that have not yet been deployed.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+        if status == "undeployed":
+            print("All nodes have been undeployed and running notify might have unexpected consequences.")
+            question = prompt_yes_no_question()
+            if not question:
+                return 0
+
+    if not args.force and not args.trigger:
+        print("You have not specified which policy trigger to use (with --trigger/-t or --event/-e) "
+              "and in this case all the triggers will be invoked which might not be what you want.")
+        question = prompt_yes_no_question()
+        if not question:
+            return 0
```

Comment:
I would make the trigger parameter mandatory. It's much easier to later make it optional than the other way around.

Reply:
Is there a case where someone would want and _need_ to do that? In the context of scaling, I feel some concurrency issues will show up with running multiple scaling operations at the same time. I'd solve that case with something like

```bash
opera notify --trigger "$(opera notify --list)"
```

to trigger many at once, but that's something we'd add later. I suggest keeping things as simple as possible until we know how we actually use `notify`, because I have no clue myself :sweat_smile:

Reply:
Yes, running multiple triggers that should not be run together on the same node could be a problem (and the user is warned about that). As you put it, invoking all the triggers probably would not be so common, it would be sensible when having different types of nodes (e.g. OpenStack VM, AWS EC2 and so on) and you wanna scale them all at once.

I agree that we don't know yet how `opera notify` will be used (apart from SaaS). So, till then I'm making `--trigger` param mandatory and if we find it problematic we can still make it optional in the future.